extmod/mpbthci.h

```diff
@@ -28,6 +28,8 @@
 #define MICROPY_INCLUDED_EXTMOD_MPBTHCI_H
 
 // --- Optionally can be implemented by the driver. ---------------------------
+typedef void (*mp_bluetooth_hci_uart_readchar_t)(uint8_t chr);
```

Comment:
Maybe rename to `mp_bluetooth_hci_uart_packet_byte_handler_t` and only `#if MICROPY_PY_BLUETOOTH_HCI_READ_MODE_PACKET` and group with the decl of `mp_bluetooth_hci_uart_readpacket`.